File: 27.py
from typing import List
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        p1 = 0
        p2 = 0
        k = 0

        while p1 <= len(nums) - 1 and p2 <= len(nums) - 1:
            if nums[p1] != val:
                nums[p2] = nums[p1]
                p2 += 1
                p1 += 1
                k += 1
            else:
                p1 += 1

        return k





nums = [3, 2, 2, 3]
val = 3
sol = Solution()
print(sol.removeElement(nums, val))

File: 27.cpp
//
// Created by DuLi on 2023/9/2.
//
#include <iostream>
using std::vector;

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int k = 0;
        for (int i = 0; i < nums.size(); ++i) {
            if (nums[i] != val){
                nums[k] = nums[i];
                k++;
            }
        }
        return k;
    }
};

int main(){
    Solution sol;
    vector<int> nums = {3, 2, 2, 3};
    int val = 2;
    int result = sol.removeElement(nums, val);
    std::cout << result;
}